src/healthcheck.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import delay from 'delay'
+import { RTMClient } from '@slack/client'
+import config from 'config'
+import dbApi from './db-api'
+import type { DbApi } from 'icarus-backend'; // eslint-disable-line
+
+const { logger } = config.get('server')
+
+async function fetchBestBlock(db) {
+  return dbApi(db).bestBlock()
+}
+
+async function start(db: any) {
+  console.log('start') // eslint-disable-line
+  const token = process.env.SLACK_TOKEN
+  const channelId = process.env.SLACK_CHANNEL
+  process.env.BLOCK_RESPONSE = 'true'
+  const rtm = new RTMClient(token)
+  rtm.start()
+
+  let responding = true
+  let bestBlock = await fetchBestBlock(db)
+
+  while (true) { // eslint-disable-line
+    await delay(70000) // eslint-disable-line
+    const dbBestBlock = await fetchBestBlock(db) // eslint-disable-line
+    const changed = !(bestBlock === dbBestBlock)
+
+    if (responding !== changed) {
+      /*
+        If the block did not change, it is guaranteed that the database does not contain the
+        latest data. However, it is possible that the database does not contain the latest data
+        even if the block did change, for example in the case when the database is synchronising
+        with the blockchain.
```

Comment:
I would change the "However..." into "TODO: assure that the server stops serving data even in case the current best block has changed but it is too old (e.g. when restoring the database and it is not ready yet)"

Reply:
hmm, I have an idea how to deal with the recovering state of the db. We can get the expected block number from the official cardano explorer - https://cardanoexplorer.com/api/blocks/pages - it would be the (# of pages * 10 + # of items on the last page) the number of pages is the first item returned. But this check should be performed of course only if the request succeeds, otherwise it should fall back to the block number change check which we currently have. Probably both of the checks should be performed so even if the official explorer stops syncing, we would still discover that our block number is not increasing.